asrtoolkit/wer.py

```diff
@@ -41,11 +41,16 @@ def remove_nonsilence_noises(input_text):
     return re.sub(re_nonsilence_noises, "", input_text)
 
 
-def wer(ref, hyp, remove_nsns=False):
+def wer(ref, hyp, remove_nsns=False, return_counts=False):
```

Comment:
```suggestion
def wer_counts(ref, hyp, remove_nsns=False):
    """
    Calculate word error between two string or time_aligned_text objects as tuple.
    The first number is the number of errors, the second is the number of words.
    >>> wer("this is a cat", "this is a dog")
    (1, 4)
    """

    # accept time_aligned_text objects too
    if type(ref) == time_aligned_text:
        ref = ref.text()

    if type(hyp) == time_aligned_text:
        hyp = hyp.text()

    # remove tagged noises and other nonspeech events
    ref = re.sub(re_tagged_nonspeech, " ", ref)
    hyp = re.sub(re_tagged_nonspeech, " ", hyp)

    # optionally, remove non silence noises
    if remove_nsns:
        ref = remove_nonsilence_noises(ref)
        hyp = remove_nonsilence_noises(hyp)

    # clean punctuation, etc.
    ref = clean_up(ref)
    hyp = clean_up(hyp)

    error_count = editdistance.eval(ref.split(" "), hyp.split(" "))
    word_count = max(1, len(ref.split(" ")))

    return error_count, word_count


def wer(ref, hyp, remove_nsns=False):
    """
    Calculate word error rate between two string or time_aligned_text objects
    >>> wer("this is a cat", "this is a dog")
    25.0
    """

    error_count, word_count = wer_counts(ref, hyp, remove_nsns)

    # calculate WER
    return 100 * error_count / word_count
```

Hey, thanks for the suggestion. What do you think about making a helper function to provide the raw numbers? That way you can just call the helper function directly if that is what you are targeting.

Reply:
I separated it out into a helper function and added to this pull request. Thoughts?